### PR TITLE
Implement merge method on view

### PR DESCRIPTION
**Pull request by jnwatson**
A new view that was forked from another view can now be merged back into the view it was forked on.

All changes from the child view are incorporated in its parent view and it is destroyed. As triggers were fired when they were applied to the child view, triggers do not fire when merged into the parent.

Limitation: a view can't merge into its parent if it has any existing forked children.

### CODE REVIEW

File: synapse/lib/view.py [hunk 1]
```diff
@@ -300,6 +300,46 @@ def pack(self):
...
+            if view.parent is not None and view.parent == self:
+                raise s_exc.SynErr(mesg='Cannot merge a view that has children itself')
+
+        CHUNKSIZE = 1000
+        fromoff = 0
+
+        if user is None:
+            user = self.core.auth.getUserByName('root')
+
+        await self.core.boss.promote('storm', user=user, info={'merging': self.iden})
```
Comment (vEpiphyte) [inline synapse/lib/view.py]: Is there a piece of book-keeping here that we need to do in order to check/kill any existing tasks which are using the view we're merging? or to lock out or set it to readonly somehow?

File: synapse/lib/view.py [hunk 2]
```diff
@@ -300,6 +300,46 @@ def pack(self):
...
+        async with await self.parent.snap(user=user) as snap:
+            snap.disableTriggers()
+            snap.strict = False
+            while True:
+                splicechunk = [x async for x in fromlayr.splices(fromoff, CHUNKSIZE)]
+
+                await snap.addFeedData('syn.splice', splicechunk)
```
Comment (vEpiphyte) [inline synapse/lib/view.py]: Are we defering the set difference view for now?

File: synapse/lib/view.py (hunk 1 above)
Comment (jnwatson) [reply synapse/lib/view.py]: We're deleting the view. I think there's an open issue re: what to do about tasks using an deleted view, but I think that's out of scope.

File: synapse/lib/view.py (hunk 2 above)
Comment (jnwatson) [reply synapse/lib/view.py]: Discussed offline

**Pull request merged**